misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/PerformanceAnalyzerService.java

```diff
@@ -88,4 +101,68 @@ public void init(LanguageServer langServer, WorkspaceManager workspaceManager) {
         });
     }
 
+    @JsonNotification
+    public CompletableFuture<List<PerformanceAnalyzerResponse>> getResourcesWithEndpoints(
+            PerformanceAnalyzerRequest request) {
+
+        return CompletableFuture.supplyAsync(() -> {
+            List<PerformanceAnalyzerResponse> resourcesWithEndpoints = new ArrayList<>();
+
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Path path = Path.of(fileUri);
+
+            Optional<SemanticModel> semanticModel = this.workspaceManager.semanticModel(path);
+            Optional<Module> module = this.workspaceManager.module(path);
+
+            if (semanticModel.isEmpty() || module.isEmpty()) {
+                return resourcesWithEndpoints;
+            }
+
+            ResourceFinder nodeVisitor = new ResourceFinder();
+
+            Optional<Document> document = this.workspaceManager.document(path);
+            if (document.isEmpty()) {
+                return resourcesWithEndpoints;
+            }
+
+            SyntaxTree syntaxTree = document.get().syntaxTree();
+            syntaxTree.rootNode().accept(nodeVisitor);
+            List<Resource> resourceRanges = nodeVisitor.getResources();
+
+            for (Resource resource : resourceRanges) {
+
+                LineRange range = resource.getLineRange();
+                Range lineRange = new Range(new Position(range.startLine().line(), range.startLine().offset()),
+                        new Position(range.endLine().line(), range.endLine().offset()));
+
+                PerformanceAnalyzerResponse response = new PerformanceAnalyzerResponse();
+                response.setName(resource.getName());
+                response.setResourcePos(lineRange);
+
+                JsonObject data = EndpointsFinder.getEndpoints(fileUri, this.workspaceManager, lineRange);
```

Comment:
Since this is a new API, what if we use a Bean, instead of `JsonObject?` That will be better in the API point of view since we can track the changes to the API